File: aula2/credit.c
#include <stdio.h>
#include <cs50.h>

int sum_digits(int n) 
{
    int sum = 0;
    while (n > 0)
    {
        sum += (n - (n / 10) * 10);
        n = n / 10;
    }
    return sum;
}

int main(void)
{
    long int card_number = get_long("Number: ");
    long int aux_number = card_number;
    int length = 0;
    int first_digit, second_digit;
    while (aux_number > 0)
    {
        length++;
        first_digit = aux_number;
        if (aux_number > 10)
        {
            second_digit = aux_number;
        }
        aux_number = aux_number / 10;
    }
    int counter = 1;
    int sum = 0;
    while (card_number > 0)
    {
        if (counter % 2 == 0)
        {
            sum += sum_digits((card_number - (card_number / 10) * 10) * 2);
            card_number = card_number / 10;
        }
        else
        {
            sum += sum_digits(card_number - (card_number / 10) * 10);
            card_number = card_number / 10;
        }
        counter++;
    }
    if (sum % 10 != 0)
    {
        printf("INVALID\n");
    }
    else
    {
        if (first_digit == 4 && (length == 13 || length == 16))
        {
            printf("VISA\n");
        }
        else if (first_digit == 5 && (second_digit >= 51 && second_digit <= 55) && length == 16)
        {
            printf("MASTERCARD\n");
        }
        else if (first_digit == 3 && (second_digit == 34 || second_digit == 37) && length == 15)
        {
            printf("AMEX\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
}

File: aula7/credit.py
def sum_algorisms(n):
    return sum([int(algo) for algo in str(n)])


while True:
    try:
        card = input("Number: ")
        card_sum = 0
        for index in range(len(card) % 2, len(card) - 1, 2):
            card_sum += sum_algorisms(int(card[index]) * 2)
        for index in range(1 - len(card) % 2, len(card), 2):
            card_sum += int(card[index])

        if (card_sum % 10 != 0):
            print("INVALID")
        else:
            if (len(card) == 15 and (card[0] == '3' and (card[1] == '4' or card[1] == '7'))):
                print('AMEX')
            elif (len(card) == 16 and (card[0] == '5' and (int(card[1]) >= 1 and int(card[1]) <= 5))):
                print('MASTERCARD')
            elif ((len(card) == 13 or len(card) == 16) and card[0] == '4'):
                print('VISA')
            else:
                print('INVALID')
        break
    except:
        continue
